File: CS.NET/PdfViewerCSharpAPI_R2/DocumentManagement/Requests/PdfGetTextFragmentsRequest.cs
// -----------------------------------------------------------------------
// <copyright file="PdfGetPageRangeRequest.cs" company="">
// TODO: Update copyright text.
// </copyright>
// -----------------------------------------------------------------------

namespace PdfTools.PdfViewerCSharpAPI.DocumentManagement.Requests 
{
    using System;
    using System.Collections.Generic;
    using System.Collections.Concurrent;
    using System.Linq;
    using System.Text;
    using PdfTools.PdfViewerCSharpAPI.Utilities;
    using PdfTools.PdfViewerCSharpAPI.Model;

    /// <summary>
    /// TODO: Update summary.
    /// </summary>
    public class PdfGetTextFragmentsRequest : APdfRequest<int, IList<PdfTextFragment>>
    {
        public PdfGetTextFragmentsRequest(int page)
            : base(page, 40)
        { }

        protected override IList<PdfTextFragment> ExecuteNative(IPdfDocument document, int page)
        {
            return document.LoadTextFragments(page);
        }

        protected override void triggerControllerCallback(IPdfControllerCallbackManager controller, InOutTuple tuple, PdfViewerException ex)
        {
        }
        protected override void triggerControllerCallback(IPdfControllerCallbackManager controller, PdfViewerException ex)
        {
        }
    }
}

File: Java/PdfViewerAPI_R2/java/com/pdf_tools/pdfviewer/Requests/PdfGetTextFragmentsRequest.java
package com.pdf_tools.pdfviewer.Requests;

import java.util.List;

import com.pdf_tools.pdfviewer.DocumentManagement.IPdfDocument;
import com.pdf_tools.pdfviewer.Model.IPdfControllerCallbackManager;
import com.pdf_tools.pdfviewer.Model.PdfTextFragment;
import com.pdf_tools.pdfviewer.Model.PdfViewerException;

public class PdfGetTextFragmentsRequest extends APdfRequest<List<PdfTextFragment>>
{

    public PdfGetTextFragmentsRequest(int pageNo)
    {
        this._priority = 82;
        this.pageNo = pageNo;
    }

    @Override
    public void execute(IPdfDocument document, IPdfControllerCallbackManager controller)
    {
        try
        {
            List<PdfTextFragment> frags = document.getTextFragments(pageNo);
            completedEvent.triggerEvent(frags, null);
        } catch (PdfViewerException ex)
        {
            completedEvent.triggerEvent(null, ex);
        }
    }

    private int pageNo;

}
